StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1027CodeFixProvider.cs

```diff
@@ -49,53 +49,84 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
         {
             var indentationOptions = IndentationOptions.FromDocument(document);
-            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
-            var violatingTrivia = syntaxRoot.FindTrivia(diagnostic.Location.SourceSpan.Start);
-
-            var stringBuilder = new StringBuilder();
+            SourceText sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            return document.WithText(sourceText.WithChanges(FixDiagnostic(indentationOptions, sourceText, diagnostic)));
+        }
 
-            int firstTriviaIndex = violatingTrivia.GetLineSpan().StartLinePosition.Character;
-            string relevantText;
-            if (firstTriviaIndex == 0)
-            {
-                relevantText = violatingTrivia.ToFullString();
-            }
-            else
-            {
-                SourceText sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-                relevantText = sourceText.ToString(new TextSpan(violatingTrivia.FullSpan.Start - firstTriviaIndex, firstTriviaIndex + violatingTrivia.FullSpan.Length));
-            }
+        private static TextChange FixDiagnostic(IndentationOptions indentationOptions, SourceText sourceText, Diagnostic diagnostic)
+        {
+            TextSpan span = diagnostic.Location.SourceSpan;
 
+            TextLine startLine = sourceText.Lines.GetLineFromPosition(span.Start);
+            string text = sourceText.ToString(TextSpan.FromBounds(startLine.Start, span.End));
+            StringBuilder replacement = new StringBuilder();
```

Comment:
Why not to pass `text.Length` as capacity? I know that resulting string will be larger than `text.Length` but anyway it will reduce amount of SB chunks.